File: src/main/java/itmo/is/project/dto/module/BuildModuleRequest.java
package itmo.is.project.dto.module;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.v3.oas.annotations.media.Schema;

public record BuildModuleRequest(

        @Schema(example = "1")
        @JsonProperty(value = "blueprintId", required = true)
        Integer blueprintId
) {
}

File: src/main/kotlin/itmo/isproject/shared/module/dto/BuildModuleRequest.kt
package itmo.isproject.shared.module.dto

import com.fasterxml.jackson.annotation.JsonProperty
import io.swagger.v3.oas.annotations.media.Schema

data class BuildModuleRequest(

    @Schema(example = "1")
    @JsonProperty(value = "blueprintId", required = true)
    val blueprintId: Int
)
